domene/src/main/kotlin/no/nav/tiltakspenger/saksbehandling/domene/vilkår/kvp/KVPVilkår.kt

```diff
@@ -0,0 +1,93 @@
+package no.nav.tiltakspenger.saksbehandling.domene.vilkår.kvp
+
+import no.nav.tiltakspenger.libs.periodisering.Periode
+import no.nav.tiltakspenger.libs.periodisering.PeriodeMedVerdi
+import no.nav.tiltakspenger.libs.periodisering.Periodisering
+import no.nav.tiltakspenger.saksbehandling.domene.vilkår.Lovreferanse
+import no.nav.tiltakspenger.saksbehandling.domene.vilkår.SkalErstatteVilkår
+
+import no.nav.tiltakspenger.saksbehandling.domene.vilkår.Utfall2
+
+/**
+ * Kvalifiseringsprogrammet (KVP): https://www.nav.no/kvalifiseringsprogrammet
+ *
+ * @param søknadSaksopplysning Saksopplysninger som kan være avgjørende for vurderingen. Kan ikke ha hull. [avklartSaksopplysning]/faktumet er den avgjørende saksopplysningen.
+ * @param avklartSaksopplysning Faktumet som avgjør om vilkåret er oppfylt eller ikke. Null implisiserer uavklart.
+ * @param utfall Selvom om utfallet er
+ *
+ */
+data class KVPVilkår private constructor(
+    val søknadSaksopplysning: KvpSaksopplysning,
+    val saksbehandlerSaksopplysning: KvpSaksopplysning?,
+    val avklartSaksopplysning: KvpSaksopplysning,
+    val utfall: Periodisering<Utfall2>,
+) : SkalErstatteVilkår {
+
+    val totaleUtfall: Utfall2 = when {
+        utfall.perioder().any { it.verdi == Utfall2.UAVKLART } -> Utfall2.UAVKLART
+        utfall.perioder().all { it.verdi == Utfall2.OPPFYLT } -> Utfall2.OPPFYLT
+        utfall.perioder().all { it.verdi == Utfall2.IKKE_OPPFYLT } -> Utfall2.IKKE_OPPFYLT
+        else -> throw IllegalStateException("Ugyldig utfall")
```

Comment:
Hvis noen av periodene har `Utfall2.OPPFYLT` og noen har `Utfall2.IKKE_OPPFYLT` får man `IllegalStateException` da? For det er delvis innvilget? Og det skal vi ikke støtte enda? 🤔
Har pleid å ha en `IkkeImplementertEndaException` for sånne scenarioer, men det er smak og behag 😊 Liker noen form for `IkkeImplementertEndaException` for da blir det lett å se på feilmeldingen at det er tenkt funksjonalitet

Reply:
Endret denne til SamletUtfall, der man støtter DELVIS_OPPFYLT

Reply:
Tanken var ikke at dette ikke var implementert.